### DISCUSSION TITLE
How do I opt out of fetch cache without making dynamic routes?

### DISCUSSION THREAD

**Discussion by alfonsusac** [vercel/next.js#54496] [Help]
### Summary

How do I opt out of fetch cache, i.e using `{ cache: no-store }` without turning my route into runtime server side render a.k.a lambda? I want to fetch something large, more than 2MB cache limit, but only want to be run at initial static build time (static generation).

The temporary solution is just to remove `{ cache: 'no-store" }` which will make fetch try cache the 4MB result but logs an error. It will build successfully but the error still persists which can be ignored. Also applies to dev server, error occurs but can be ignored.

It seems weird that you can't opt out of fetch caching without making the route dynamically rendered at request time. Is there a reason why this is the case?

### Additional information

```js
export const dynamic = "error";
export default function Home() {
  const url = "...";
  const img = fetch(url, { cache: "no-store" });

  return <main></main>;
}
```

Result:
- No error on dev, but console.log prints at request time
- Error on build

I have tried using "force-static" which seems to not give any effect, the route will be computed at request time at both dev and prod despite the declared config.

I have tried downgrading and the error still persists. 

Next Info
```
    Operating System:
      Platform: win32
      Arch: x64
      Version: Windows 10 Pro
    Binaries:
      Node: 18.15.0
      npm: N/A
      Yarn: N/A
      pnpm: N/A
    Relevant Packages:
      next: 13.4.20-canary.4
      eslint-config-next: 13.4.19
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 5.1.6
    Next.js Config:
      output: N/A
```

**Comment by alfonsusac** [comment]
### Temporary Solution

For some reason I've found out that you can override the default Next.js fetching behavior by installing `node-fetch`.

```ts
import fetch2 from 'node-fetch'
export const dynamic = "error";
export default function Home() {
  const url = "...";
  const img = fetch2(url);

  return <main></main>;
}
```

**Comment by evidentsnow** [comment]
Also having this problem
